_rules/widget-non-empty-accessible-name-rdzs6q.md

```diff
@@ -0,0 +1,377 @@
+---
+id: rdzs6q
+name: Widget has non-empty accessible name
+rule_type: atomic
+description: |
+  This rule checks that each widget has a non-empty accessible name.
+accessibility_requirements:
+  wcag20:4.1.2: # Name, Role, Value (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - Accessibility Tree
+  - CSS styling
+  - DOM Tree
+acknowledgments:
+  authors:
+    - Carlos Duarte
+  previous_authors:
+    - Anne Thyme Nørregaard
+    - Bryn Anderson
+    - Stein Erik Skotkjerra
+    - Wilco Fiers
+  assets:
+    - Image used in passed example 14 is courtesy of NASA/JPL-Caltech.
+htmlHintIgnore:
+  # https://www.npmjs.com/package/htmlhint
+  # (used with `npm test` to ensure validity of code snippets)
+  - 'alt-require'
+---
+
+## Applicability
+
+This rule applies to any element that is [included in the accessibility tree](#included-in-the-accessibility-tree) and has a [semantic role][] that is one of the following [semantic roles][]:
+
+- `button`; or
+- `checkbox`; or
+- `combobox`; or
+- `link`; or
+- `listbox`; or
+- `menuitem`; or
+- `menuitemcheckbox`; or
+- `menuitemradio`; or
+- `radio`; or
+- `searchbox`; or
+- `slider`; or
+- `spinbutton`; or
+- `switch`; or
+- `textbox`.
+
+## Expectation
+
+Each target element has an [accessible name][] that is not empty (`""`).
+
+## Assumptions
+
+_There are currently no assumptions_
+
+## Accessibility Support
+
+- Certain assistive technologies can be set up to ignore the title attribute, which means that to some users the title attribute will not act as an [accessible name][].
+- Implementation of [Presentational Roles Conflict Resolution][] varies from one browser or assistive technology to another. Depending on this, some elements can have one of the applicable [semantic roles][] and fail this rule with some technology but users of other technologies would not experience any accessibility issue.
+
```

Comment:
We should probably keep the second point of https://www.w3.org/WAI/standards-guidelines/act/rules/button-non-empty-accessible-name-97a4e1/#accessibilitysupport explaining why we accept "submit" and "reset", especially since that rule is deprecated.

Reply:
I'm wandering... given that we are not deprecating the "image button" rule, do we need to fail image buttons that have the default name here? Or is it enough to fail them in the "image button" rule?
This rule would just check the "non-emptyness" of the accessible name, without judging whether the default names are appropriate or not...

Reply:
That's a fair point.
`<input type="image" />` will have an accessible name of "Submit Query" and fails both 1.1.1 and 4.1.2 via "Image button has non-empty accessible name". Having it pass this rule creates a false negative for 4.1.2 but makes the rule so much simpler that it is totally worth it (vs "accessible name is not empty nor the default name of image button" which would be awkward).

Then I guess just a background note explaining this (and why one rule accepts "Submit Query" while the other rejects it).